hathor/p2p/sync_v2/agent.py

```diff
@@ -545,36 +549,42 @@ def find_best_common_block(self,
         # Run an n-ary search in the interval [lo, hi).
         # `lo` is always a height where we are synced.
         # `hi` is always a height where sync state is unknown.
-        hi = min(peer_best_block.height, my_best_block.height)
-        lo = 0
-
-        lo_block_hash = self._settings.GENESIS_BLOCK_HASH
+        hi = min(peer_best_block, my_best_block)
```

Comment:
**issue:** Using `min` on a `NamedTuple` is error prone.

I suggest using the `key` argument to select the block height, just like it's done in the `sort` below. Otherwise, this hinges on a specific ordering in the named tuple.

Reply:
Fixed! Thanks!

Reply:
Thanks!